tensorrt_llm/_torch/visual_gen/attention_backend/parallel.py

```
@@ -714,7 +760,122 @@ def _ring_send_recv(self, send: torch.Tensor, recv: torch.Tensor) -> None:
+            padded[: len(local_desc)] = desc_tensor
+            all_descs = [torch.zeros(max_len, dtype=torch.uint8).cuda() for _ in range(self.world_size)]
+            dist.all_gather(all_descs, padded, group=self.pg)
+            ring_rank = dist.get_rank(group=self.pg)
+            prev_ring_rank = (ring_rank - 1) % self.world_size
+            prev_global_rank = dist.get_global_rank(self.pg, prev_ring_rank)
+            prev_len = int(all_lens[prev_ring_rank].item())
+            prev_desc = bytes(all_descs[prev_ring_rank][:prev_len].cpu().tolist())
+            self._nixl_prev_agent_name = f"ring_rank_{prev_global_rank}"
+            self._nixl_agent.load_remote_agent(self._nixl_prev_agent_name, prev_desc)
+            self._nixl_ready = True
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)